### PR TITLE
feature: add `FluidXScanner`-class

### PR DESCRIPTION
**Pull request by simbig**
- [x] Added automated tests
- [x] Documented for all relevant versions
- [x] Updated the changelog

**Changes**

- Add 'FluidXScanner'- class

### CODE REVIEW

File: src/FluidXPlate/FluidXScanner.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+/**
+ * Communicates with a FluidX scanner device and fetches results from it.
+ */
+class FluidXScanner
+{
+    private const READING = 'Reading...';
+    private const XTR_96_CONNECTED = 'xtr-96 Connected';
+    private const NO_READ = 'NO READ';
+    private const NO_TUBE = 'NO TUBE';
+    public const TEST_IP = '127.0.0.1';
```
Comment (spawnia) [inline src/FluidXPlate/FluidXScanner.php]:
```suggestion
    public const LOCALHOST = '127.0.0.1';
```

File: src/FluidXPlate/FluidXScanner.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+    public function scanPlate(string $ip): FluidXPlate
+    {
+        if (self::TEST_IP === $ip) {
+            return self::returnTestPlate();
+        }
+
+        if ('' === $ip) {
+            throw new ScanFluidXPlateException('Cannot start scan request without a IP address');
```
Comment (spawnia) [inline src/FluidXPlate/FluidXScanner.php]:
```suggestion
            throw new ScanFluidXPlateException('Cannot start scan request without an IP address.');
```

File: src/FluidXPlate/FluidXScanner.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            throw new ScanFluidXPlateException('Der Scanner lieferte ein leeres Ergebnis zurück.');
+        }
+
+        $lines = StringUtil::splitLines($rawContent);
+        $barcodes = [];
+        $id = null;
+        foreach ($lines as $line) {
+            if ('' !== $line
+                && self::READING !== $line
+                && self::XTR_96_CONNECTED !== $line
```
Comment (spawnia) [inline src/FluidXPlate/FluidXScanner.php]: In this case `continue`?

File: src/FluidXPlate/FluidXScanner.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            $plate->addWell(Coordinate::fromString($coordinate, new CoordinateSystem96Well()), $barcode);
+        }
+
+        if (FluidXScanner::NO_READ === $id) {
+            if ([] === $barcodes) {
+                throw new ScanFluidXPlateException(
+                    'Weder Platten-Barcode noch Tube-Barcodes konnten gescannt werden. Bitte überprüfen Sie, dass die Platte korrekt in den FluidX-Scanner eingelegt wurde.'
+                );
+            }
+            throw new ScanFluidXPlateException('Platten-Barcode konnte nicht gescannt werden. Bitte überprüfen Sie, dass die Platte mit der korrekten Orientierung in den FluidX-Scanner eingelegt wurde.');
```
Comment (spawnia) [inline src/FluidXPlate/FluidXScanner.php]:
```suggestion
            throw new ScanFluidXPlateException($barcode === []
                ? 'Weder Platten-Barcode noch Tube-Barcodes konnten gescannt werden. Bitte überprüfen Sie, dass die Platte korrekt in den FluidX-Scanner eingelegt wurde.'
                : 'Platten-Barcode konnte nicht gescannt werden. Bitte überprüfen Sie, dass die Platte mit der korrekten Orientierung in den FluidX-Scanner eingelegt wurde.');
```

File: tests/Unit/FluidXPlate/FluidXScannerTest.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+final class FluidXScannerTest extends TestCase
+{
+    public function testCreateFromStringEmpty(): void
+    {
+        $fluidXScanner = new FluidXScanner();
+        $fluidXPlate = $fluidXScanner->scanPlate(FluidXScanner::TEST_IP);
+
+        self::assertSame('SA00826894', $fluidXPlate->rackId);
+        $collection = $fluidXPlate->filledWells();
```
Comment (spawnia) [inline tests/Unit/FluidXPlate/FluidXScannerTest.php]:
```suggestion
        $filledWells = $fluidXPlate->filledWells();
```

File: src/FluidXPlate/FluidXScanner.php [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        $barcodes = [];
+        $id = null;
+        foreach ($lines as $line) {
+            if ('' !== $line
+                && self::READING !== $line
+                && self::XTR_96_CONNECTED !== $line
+            ) {
+                $content = explode(', ', $line);
+
+                if (count($content) > 3) {
```
Comment (spawnia) [inline src/FluidXPlate/FluidXScanner.php]: I guess my comment was not clear. What I meant is that the flying V in this section can be reduced by using early returns (in this case `continue`).

Comment (simbig) [reply src/FluidXPlate/FluidXScanner.php]: done

**Review by spawnia — APPROVED** [review]

**Pull request merged**